### PR TITLE
Notebook rendering and linting

### PR DESCRIPTION
**Pull request by CBroz1**
### Submission checklist

- [x] Reviewed all `Files changed` prior to selecting `Create Pull Request`.
- [ ] Ran and passed all pytests, and ran the pipeline for features not covered by tests.
- [X] Ran pre-commit checks, including black, flake8, etc.
- [X] Updated the changelog and if making a new release, updated the version.py. **Proposing no new release**
- [X] Updated documentation, including concepts.md, tutorials.md, and relevant docstrings.
- [X] After merge, I will push a new tag to upstream (test* prefix if no new release).

### Submission details

In this pull request, I make the following changes:
- Concept: Add linters
  - `markdownlint.yaml`: config for markdown linting
  - `pre-commit-config.yaml`: config for pre-commit, which enforces various linting checks including whitespace, black, flake8 and sorting imports
  - `cspell.json`: Add words to default dictionaries.
- Add docs tutorial subfolder and support for notebook rendering
  - `gitignore`: ignore notebooks in commits
  - `docs/.docker/pip_requirements` - add jupyter extension
  - `docs/docker-compose` - add download of notebook
  - `make_pages.py` - remove workflow download subprocess, migrate to docker-compose
  - `mkdocs.yaml` - add routing for new notebook and tutorial index files
  - remove `tutorial.md` and replace with `tutorials/index.md`
- Add typing suggestions for API docs rendering. Particularly `->` returns types.
  - `cell_plot.py`
  - `widget.py`
- Enforce linting 
  - fix whitespace and trailing characters: 
    - `gitignore`
  - markdownlint `-` for unordered lists
    - `citation.md`
    - `concepts.md`
  - markdown wrap at 88: 
    - `CHANGELOG`
    - `CONTRIBUTING`
    - `README`
    - `citation.md`
    - `index.md`
  - remove/add spacing around markdown headings: 
    - `concepts.md`
  - Spelling corrections: 
    - `mkdocs.yaml`
    - `concepts.md`
    - `imaging.py`
    - `imaging_no_curation.py`
    - `imaging_preprocess.py`
    - `imaging_report.py`
    - `scan.py`
  - reorder imports via isort:
    - `imaging.py`
    - `imaging_no_curation.py`
    - `imaging_preprocess.py`
    - `imaging_report.py`
    - `scan.py`
    - `setup.py`
  - Flake8 issues:
    - `setup.py` - ignore F821 error on `__version__`
    - `scan.py` - ignore E722 bare `except` statements pending future PRs
    - `imaging_preprocess.py` - set unused variable name to `_`
    - Change f-strings to bare strings where no variable is used
  - Apply `black`
- Misc minor fixes:
  - `gitignore`: ignore temp* prefixed files including new mkdocs log
  - update `LICENSE` to 2023
  - `docs/docker-compose` - add `tee` statement for logging mkdocs build to temp file for debugging docs build
  - `mkdocs.yaml` - change api rendering to respect source ordering

**NOTE**: Marked draft pending resolution of issues with Explore notebook unicode strings that interrupt conversion.

### CODE REVIEW

**Comment by CBroz1** [comment]
Hi @kabilar - If we're ready to move forward with this docs update without notebook 5, this PR is ready for review. I'll plan to address this notebook 5 issue after the dev container is in place

File: .github/ISSUE_TEMPLATE/config.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+blank_issues_enabled: false
+contact_links:
+  - name: DataJoint Contribution Guideline
+    url: https://docs.datajoint.org/python/community/02-Contribute.html
```
Comment (kabilar) [inline .github/ISSUE_TEMPLATE/config.yml:4]:
```suggestion
    url: https://datajoint.com/docs/community/contribute/
```

File: docs/mkdocs.yaml [hunk 2]
```diff
@@ -7,7 +7,17 @@ repo_name: datajoint/element-calcium-imaging
...
+    - Overview: tutorials/index.md
+    - Data Download: tutorials/00-data-download-optional.ipynb
+    - Configure:  tutorials/01-configure.ipynb
+    - Workflow Structure: tutorials/02-workflow-structure-optional.ipynb
+    - Process:  tutorials/03-process.ipynb
+    - Automate:  tutorials/04-automate-optional.ipynb
+    # - Explore:  tutorials/05-explore.ipynb
+    - Drop:  tutorials/06-drop-optional.ipynb
+    - Downstream Analysis: tutorials/07-downstream-analysis-optional.ipynb
+    - Workshop Demo: tutorials/2022-allen-institute-workshop.ipynb
```
Comment (kabilar) [inline docs/mkdocs.yaml:20]:
```suggestion
```

Comment (kabilar) [reply docs/mkdocs.yaml:20]: Let's keep this notebook in the repo, but not render on the docs site.

**Comment by kabilar** [comment]
The `cell_plot.py` and `widget.py` files are showing that all lines changed.  Is there any way to see a simpler diff of just the few lines that changed?

**Comment by kabilar** [comment]
> That's odd. I'm seeing a relatively sparse diff, as shown below. Is that not what you see?
> 
> For these files, I sort imports via `isort` and add `-> output` typing suggestions to support API docs rendering. I make no functional changes.

Thanks.  Those changes look good.  For me it shows that all lines changed between the files.

File: docs/src/tutorials/index.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+complete pipeline, and is able to ingest experimental metadata and process calcium
+imaging scans.
+
+## Videos
+
+Our [YouTube tutorial](https://www.youtube.com/watch?v=gFLn0GB1L30) gives an overview of
+the workflow files, notebooks, as well as core concepts related to calcium imaging
+analysis. To try out Elements notebooks in an online Jupyter environment with access to
+example data, visit
+[CodeBook](https://codebook.datajoint.io/). (Calcium Imaging notebooks coming soon!)
```
Comment (kabilar) [inline docs/src/tutorials/index.md:21]:
```suggestion
Our [YouTube tutorial](https://www.youtube.com/watch?v=gFLn0GB1L30) gives an overview of
the workflow files and notebooks, as well as core concepts related to calcium imaging
analysis. To try out the tutorial notebooks in an online Jupyter environment with access to
example data, visit
[CodeBook](https://codebook.datajoint.io/).
```

File: docs/src/tutorials/index.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  the tables and plot them. -->
+
+- [Drop](./06-drop-optional.ipynb) provides the steps for dropping all the tables to
+  start fresh.
+
+- [Downstream Analysis](./07-downstream-analysis-optional.ipynb) demonstrates event- and
+  trial-based analysis.
+
+- [Workshop Demo](./2022-allen-institute-workshop.ipynb) provides a brief overview of
+  all of the above for a workshop setting.
```
Comment (kabilar) [inline docs/src/tutorials/index.md:59]:
```suggestion
```

File: docs/src/tutorials/index.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+[CodeBook](https://codebook.datajoint.io/). (Calcium Imaging notebooks coming soon!)
+
+## Notebooks
+
+Each of the notebooks in the workflow
+([download here](https://github.com/datajoint/workflow-calcium-imaging/tree/main/notebooks)
+steps through ways to interact with the Element itself. For convenience, these notebooks
+are also rendered as part of this site. To try out the Elements notebooks in an online
+Jupyter environment with access to example data, visit
+[CodeBook](https://codebook.datajoint.io/).
```
Comment (kabilar) [inline docs/src/tutorials/index.md:30]:
```suggestion
are also rendered as part of this site. To try out the notebooks in an online
Jupyter environment with access to example data, visit
[CodeBook](https://codebook.datajoint.io/).
```

File: docs/src/tutorials/index.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
+- [Data Download](./00-data-download-optional.ipynb) highlights how to use DataJoint
+  tools to download a sample model for trying out the Element.
+
+- [Configure](./01-configure.ipynb) helps configure your local DataJoint installation to
+  point to the correct database.
+
+- [WorkflowStructure](./02-workflow-structure-optional.ipynb) demonstrates the table
+  architecture of the Element and key DataJoint basics for interacting with these
+  tables.
```
Comment (kabilar) [inline docs/src/tutorials/index.md:40]:
```suggestion
- [Workflow Structure](./02-workflow-structure-optional.ipynb) demonstrates the table
  architecture of the Element and key DataJoint basics for interacting with these
  tables.
```

File: docs/src/tutorials/index.md [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+
+Each of the notebooks in the workflow
+([download here](https://github.com/datajoint/workflow-calcium-imaging/tree/main/notebooks)
+steps through ways to interact with the Element itself. For convenience, these notebooks
+are also rendered as part of this site. To try out the Elements notebooks in an online
+Jupyter environment with access to example data, visit
+[CodeBook](https://codebook.datajoint.io/).
+
+- [Data Download](./00-data-download-optional.ipynb) highlights how to use DataJoint
+  tools to download a sample model for trying out the Element.
```
Comment (kabilar) [inline docs/src/tutorials/index.md:33]:
```suggestion
- [Data Download](./00-data-download-optional.ipynb) highlights how to use DataJoint
  tools to download an example dataset to try out the Element.
```

**Comment by kabilar** [comment]
Other than linting, are there other changes to `concepts.md`?  The diff is showing lots of changes so it is difficult to review.

**Review by kabilar — CHANGES_REQUESTED** [review]
Thanks @CBroz1!  A few suggestions above.

**Comment by CBroz1** [comment]
> Other than linting and spell checks, are there other changes to `concepts.md`? The diff is showing lots of changes so it is difficult to review.

@kabilar - Sorry, I could have broken up the hard wrapping into a separate commit for easier review. In `concepts`, I change the following:

- Hard wrap to 88
- `+` -> `-` for bulleted lists
- Add blank lines before and after headings and lists
- Set image as in-line text rather than bullet ([here](https://github.com/CBroz1/element-calcium-imaging/blame/main/docs/src/concepts.md#L130))
- Spelling corrections for `acquisition` and `performed`

**Review by kabilar — APPROVED** [review]
Thanks @CBroz1!

**Pull request merged**